### PR TITLE
Bump app-editors/neovim 0.3.8

### PR DESCRIPTION
**Pull request by lovesegfault**
Closes: https://bugs.gentoo.org/691624
Package-Manager: Portage-2.3.71, Repoman-2.3.17
Signed-off-by: Bernardo Meurer <bernardo@standard.ai>

### CODE REVIEW

**Comment by lovesegfault** [comment]
Rebased

**Comment by XVilka** [comment]
While this still in limbo 0.4 release will be ready in a few days: https://github.com/neovim/neovim/milestone/21

**Comment by lovesegfault** [comment]
@XVilka Yeah, I've already done the legwork to support 0.4 (https://github.com/gentoo/gentoo/pull/12864)

**Comment by Tranquility** [comment]
Yes, it seems like only the libvterm update needs to be done for for 0.4... I should be able to merge this pr this weekend. I am sorry this was waiting for so long.

And hopefully will have time to prepare 0.4 too.

**Comment by XVilka** [comment]
Now 0.4.3 was released a few days ago, maybe it make sense to spend the time on that release instead?

**Comment by lovesegfault** [comment]
@XVilka I am, I'd still like for this to be merged as well though. I'm bumping libvterm so I can submit the 0.4.2 ebuild.

File: app-editors/neovim/neovim-0.3.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils xdg-utils
+
+DESCRIPTION="Vim-fork focused on extensibility and agility."
```
Comment (mgorny) [inline app-editors/neovim/neovim-0.3.8.ebuild]: This is no proper sentence, so no `.`

File: app-editors/neovim/neovim-0.3.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	luajit? ( dev-lang/luajit:2 )
+	!luajit? (
+		dev-lang/lua:=
+		dev-lua/LuaBitOp
+	)
+	tui? (
+		dev-libs/libtermkey
+		>=dev-libs/unibilium-2.0.0:0=
+	)
+	<dev-libs/libvterm-0.1
```
Comment (mgorny) [inline app-editors/neovim/neovim-0.3.8.ebuild]: Why don't you sort them? Unconditionals first, conditionals later, by cat+pn or flag name appropriately.

File: app-editors/neovim/neovim-0.3.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	dev-lua/lpeg[luajit=]
+	dev-lua/mpack[luajit=]
+	jemalloc? ( dev-libs/jemalloc )
+	net-libs/libnsl"
+
+DEPEND="
+	${CDEPEND}
+	dev-util/gperf
```
Comment (mgorny) [inline app-editors/neovim/neovim-0.3.8.ebuild]: Those three look `BDEPEND` to me.

File: app-editors/neovim/neovim-0.3.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	virtual/libintl
+	virtual/libiconv
+	app-eselect/eselect-vi"
```
Comment (mgorny) [inline app-editors/neovim/neovim-0.3.8.ebuild]: I don't really understand why eselect-vi would be a build-time dependency.

File: app-editors/neovim/neovim-0.3.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	python? ( dev-python/neovim-python-client )
+	ruby? ( dev-ruby/neovim-ruby-client )
+	remote? ( dev-python/neovim-remote )
+	clipboard? ( || ( x11-misc/xsel x11-misc/xclip ) )"
+
+CMAKE_BUILD_TYPE=Release
+
+src_prepare() {
+	# use our system vim dir
+	sed -e '/^# define SYS_VIMRC_FILE/s|$VIM|'"${EPREFIX}"'/etc/vim|' \
```
Comment (mgorny) [inline app-editors/neovim/neovim-0.3.8.ebuild]: Use outer `"` and escape `$` when you need. That `'"` transition is really confusing.

File: app-editors/neovim/neovim-0.3.8.ebuild [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+		-i src/nvim/globals.h || die
+
+	# add eclass to bash filetypes
+	sed -e 's|*.ebuild|*.ebuild,*.eclass|' -i runtime/filetype.vim || die
```
Comment (mgorny) [inline app-editors/neovim/neovim-0.3.8.ebuild]: Was this reported upstream?

File: app-editors/neovim/neovim-0.3.8.ebuild (hunk 2 above)
Comment (lovesegfault) [reply app-editors/neovim/neovim-0.3.8.ebuild]: Fixed in 0.4.2 bump, forgot to backport

File: app-editors/neovim/neovim-0.3.8.ebuild (hunk 3 above)
Comment (lovesegfault) [reply app-editors/neovim/neovim-0.3.8.ebuild]: Fixed in 0.4.2 bump, forgot to backport

File: app-editors/neovim/neovim-0.3.8.ebuild (hunk 4 above)
Comment (lovesegfault) [reply app-editors/neovim/neovim-0.3.8.ebuild]: Fixed in 0.4.2 bump, forgot to backport

File: app-editors/neovim/neovim-0.3.8.ebuild (hunk 5 above)
Comment (lovesegfault) [reply app-editors/neovim/neovim-0.3.8.ebuild]: Done, looks much better now!

File: app-editors/neovim/neovim-0.3.8.ebuild (hunk 6 above)
Comment (lovesegfault) [reply app-editors/neovim/neovim-0.3.8.ebuild]: Fixed upstream; removed

**Comment by lovesegfault** [comment]
All fixed up, @mgorny :)

**Comment by mgorny** [comment]
```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find JeMalloc (missing: JEMALLOC_LIBRARY JEMALLOC_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindJeMalloc.cmake:47 (find_package_handle_standard_args)
  CMakeLists.txt:460 (find_package)


-- Configuring incomplete, errors occurred!

**Comment by lovesegfault** [comment]
Fixed, sorry for the oversight @mgorny :)

**Pull request closed without merging**